ofcourse/site.py

```diff
@@ -36,6 +36,9 @@
 def inject_yaml():
     with open(app_path('site.yaml')) as site_yaml:
         site_config = yaml.load(site_yaml)
+        if 'public_url' not in site_config['course']:
+            course_config = site_config['course']
+            course_config['public_url'] = os.environ['OPENSHIFT_GEAR_DNS']
```

Comment:
`OPENSHIFT_GEAR_DNS` is not a full URL (no protocol), you need to add `https://` to make security explicit.